### PR TITLE
Feature destroy route

### PR DESCRIPTION
**Pull request by YunxiAmey**
Added destroy_route() method. Also fixed a bug of create_route(), where the "route" variable was set but not used in the POST request and that led to unsuccessful route creation.

### CODE REVIEW

File: flask_mailgun.py [hunk 1]
```diff
@@ -207,21 +225,42 @@ def route_exists(self, route):
...
-        # TODO should not it update?
+        route = self._build_route()
+        # Create Route Only if it does not Exist # TODO should update?
         if self.route_exists(route):
             return None
         else:
             return requests.post(self.routepoint,
                                  auth=self.auth,
-                                 data=data)
+                                 data=route)
```
Comment (Richard-Mathie) [inline flask_mailgun.py]: Woops we need to check that data is route dict in tests

File: flask_mailgun.py [hunk 2]
```diff
@@ -207,21 +225,42 @@ def route_exists(self, route):
...
+                 "expression": "match_recipient('%(route)s@%(domain)s')" % self.__dict__,
+                 "action": [action, "stop()"]}
+
+    def get_route_id(self, route):
+        """ Get id of the route
+        Return: id of the route. None if not exist.
+        """
+        routes = self.list_routes()
+        if routes:
+            id_table = dict([(r["expression"].join(r["actions"]), r["id"]) for r in routes])
```
Comment (Richard-Mathie) [inline flask_mailgun.py]: what???

is the key `(r["expression"].join(r["actions"])` seems a bit odd, and very long strings

`key = 'action1regex1action2regex1action3.....'`

File: flask_mailgun.py [hunk 3]
```diff
@@ -84,15 +84,33 @@ def send_email(self, **kwargs):
...
+        
+        """
         # register the process_email endpoint with the flask app
         @self.app.route(dest, methods=['POST'])
         def mail_endpoint():
-            return self.process_email(request)
+            #  Note: currently there is not an eay way to remove a route from flask 
+            #  app after creation. As a result, an extra if-clause is added here to 
+            #  facilitate the `destroy_route` function. For more information see:
+            #  http://stackoverflow.com/questions/24129217/flask-delete-routes-added-with-add-url
```
Comment (Richard-Mathie) [inline flask_mailgun.py]: why are we destroying the rout in the app?

wouldn't we just restart the app?

humm....

**Comment by Richard-Mathie** [comment]
build fails because of async flakiness, 

resolving this at the moment

File: flask_mailgun.py (hunk 2 above)
Comment (YunxiAmey) [inline flask_mailgun.py]: Yes. It is simply used to setup a look-up table. Not sure if it's optimal, but inside Python the keys of a dict are simply hash digests? No matter what, we need some way to make sure that the route we are destroying is EXACTLY the one that we want.

File: flask_mailgun.py (hunk 3 above)
Comment (YunxiAmey) [inline flask_mailgun.py]: Good point. I just think sometimes users cannot simply stop the app. Suppose we want to use some factory patterns to dynamically create/destroy routes for clients without having to hacking into the server, change our codes, rerun tests etc?

**Pull request merged**